src/main/java/sg/ncl/MainController.java

```diff
@@ -2560,12 +2562,130 @@ public String rejectTeam(
+
+            switch (exceptionState) {
+                case TEAM_NOT_FOUND_EXCEPTION:
+                    log.warn("Failed to restrict team {}: Team not found", team.getId());
+                    redirectAttributes.addFlashAttribute(MESSAGE, ERROR_PREFIX + "Team not found");
+                    break;
+                case INVALID_STATUS_TRANSITION_EXCEPTION:
+                    log.warn("Failed to restrict team {}: {}", team.getId(), error.getMessage());
+                    redirectAttributes.addFlashAttribute(MESSAGE, ERROR_PREFIX + error.getMessage());
+                    break;
+                case INVALID_TEAM_STATUS_EXCEPTION:
+                    log.warn("Failed to restrict team {}: {}", team.getId(), error.getMessage());
+                    redirectAttributes.addFlashAttribute(MESSAGE, ERROR_PREFIX + error.getMessage());
+                    break;
+                case FORBIDDEN_EXCEPTION:
+                    log.warn("Failed to restrict team {}: Permission denied", team.getId());
+                    redirectAttributes.addFlashAttribute(MESSAGE, ERROR_PREFIX + PERMISSION_DENIED);
+                    break;
+                default:
+                    log.warn("Failed to restrict team {}: {}", team.getId(), exceptionState.getExceptionName());
+                    redirectAttributes.addFlashAttribute(MESSAGE, ERR_SERVER_OVERLOAD);
+            }
+            return "redirect:/admin";
+        } else {
+            // good
+            log.info("Team {} has been restricted", team.getId());
+            redirectAttributes.addFlashAttribute(MESSAGE_SUCCESS, "Team " + team.getName() + " status has been changed to " + TeamStatus.RESTRICTED.name());
+            return "redirect:/admin";
+        }
+    }
+
+    private String freeTeam(final Team2 team, RedirectAttributes redirectAttributes) throws IOException {
+        log.info("Freeing team {}", team.getId());
+
+        HttpEntity<String> request = createHttpEntityHeaderOnly();
+        ResponseEntity response = restTemplate.exchange(
+                properties.getSioTeamsStatusUrl(team.getId(), TeamStatus.APPROVED),
+                HttpMethod.PUT, request, String.class);
+        String responseBody = response.getBody().toString();
+
+        if (RestUtil.isError(response.getStatusCode())) {
+            MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
+            ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+
+            switch (exceptionState) {
+                case TEAM_NOT_FOUND_EXCEPTION:
+                    log.warn("Failed to free team {}: Team not found", team.getId());
+                    redirectAttributes.addFlashAttribute(MESSAGE, ERROR_PREFIX + "Team not found");
+                    break;
+                case INVALID_STATUS_TRANSITION_EXCEPTION:
+                    log.warn("Failed to free team {}: {}", team.getId(), error.getMessage());
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define a constant instead of duplicating this literal "Failed to free team {}: {}" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)